core/src/alpenglow_consensus/voting_loop.rs

```diff
@@ -105,6 +108,9 @@ struct VotingContext {
     vote_account_pubkey: Pubkey,
     identity_keypair: Arc<Keypair>,
     authorized_voter_keypairs: Arc<RwLock<Vec<Arc<Keypair>>>>,
+    // The BLS keypair should always change with authorized_voter_keypairs.
+    derived_bls_keypairs: HashMap<Pubkey, Arc<BLSKeypair>>,
+    my_rank: HashMap<Epoch, u16>,
```

Comment:
do we really need to cache this? Seems like unnecessary headache for cleanup/maintenance since its just a lookup